shoc: warning: variable 'in_nlev_range' set but not used

in rel mode, this is quite annoying:

```
components/eamxx/src/physics/shoc/eamxx_shoc_process_interface.hpp:92:21: warning: variable 'in_nlev_range' set but not used [-Wunused-but-set-variable]
   92 |         const Smask in_nlev_range = (range < nlev);
      |         
```

--- a/components/eamxx/src/physics/shoc/eamxx_shoc_process_interface.hpp
+++ b/components/eamxx/src/physics/shoc/eamxx_shoc_process_interface.hpp
@@ -91,7 +91,7 @@ public:
         // Inverse of Exner. In non-rel builds, assert that exner != 0 when in range before computing.
         const Spack exner = PF::exner_function(p_mid(i,k));
         const Smask nonzero = (exner != 0);
-        EKAT_KERNEL_ASSERT((nonzero || !(ekat::range<IntSmallPack>(k*Spack::n) < nlev);).all());
+        EKAT_KERNEL_ASSERT((nonzero || !(ekat::range<IntSmallPack>(k*Spack::n) < nlev)).all());
         inv_exner(i,k).set(nonzero, 1/exner);
 
         tke(i,k) = ekat::max(mintke, tke(i,k));

